compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/MoveCompanionObjectFieldsLowering.kt

```diff
@@ -83,9 +84,15 @@ private class MoveCompanionObjectFieldsLowering(val context: CommonBackendContex
         companion.declarations.removeAll { it is IrAnonymousInitializer }
     }
 
-    private fun IrClass.allFieldsAreJvmField() =
-        declarations.filterIsInstance<IrProperty>()
-            .mapNotNull { it.backingField }.all { it.hasAnnotation(JVM_FIELD_ANNOTATION_FQ_NAME) }
+    private fun shouldExposeProperty(irClass: IrClass, irProperty: IrProperty): Boolean {
+        if (!irClass.isInterface && !irClass.isAnnotationClass)
+            return true
+
+        return irProperty.backingField?.let {
+            it.hasAnnotation(JVM_FIELD_ANNOTATION_FQ_NAME) || it.correspondingPropertySymbol?.owner?.isConst == true ||
```

Comment:
Is `it.correspondingPropertySymbol?.owner` the same as `irProperty.symbol`? If so, could you just use the latter? It is not nullable and you don't have to use `== true`

Reply:
`it.corresponginPropertySymbol?.owner` should be the same thing as `irProperty`. I guess the check is in the body of the let to exclude the case of delegated properties?

Reply:
Good catch! Thanks.

No, it's just me not knowing that they are the same thing; I started this patch from somewhere else dealing with IrField and found later that they are mostly done in this phase except for these cases.